modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/DistributionZoneCfg.java

```diff
@@ -0,0 +1,187 @@
+     *
+     * @return Data nodes auto adjust scale down timeout.
+     */
+    @Nullable
+    public Integer dataNodesAutoAdjustScaleDown() {
+        return dataNodesAutoAdjustScaleDown;
+    }
+
+    /**
+     * Builder for distribution zone configuration.
+     */
+    public static class Builder {
+        /** Zone name. */
+        private String name;
+
+        /** Data nodes auto adjust timeout. */
+        private Integer dataNodesAutoAdjust;
+
+        /** Data nodes auto adjust scale up timeout. */
+        private Integer dataNodesAutoAdjustScaleUp;
+
+        /** Data nodes auto adjust scale down timeout. */
+        private Integer dataNodesAutoAdjustScaleDown;
+
+        /**
+         * The constructor.
+         */
+        public Builder() {
+
+        }
+
+        /**
+         * Sets zone name.
+         *
+         * @param name Name.
+         * @return This instance.
+         */
+        public Builder name(String name) {
+            this.name = name;
+
+            return this;
+        }
+
+        /**
+         * Sets timeout in seconds between node added or node left topology event itself and data nodes switch.
+         *
+         * @param dataNodesAutoAdjust Timeout.
+         * @return This instance.
+         */
+        public Builder dataNodesAutoAdjust(int dataNodesAutoAdjust) {
+            this.dataNodesAutoAdjust = dataNodesAutoAdjust;
+
+            return this;
+        }
+
+        /**
+         * Sets timeout in seconds between node added topology event itself and data nodes switch.
+         *
+         * @param dataNodesAutoAdjustScaleUp Timeout.
+         * @return This instance.
+         */
+        public Builder dataNodesAutoAdjustScaleUp(int dataNodesAutoAdjustScaleUp) {
+            this.dataNodesAutoAdjustScaleUp = dataNodesAutoAdjustScaleUp;
+
+            return this;
+        }
+
+        /**
+         * Sets timeout in seconds between node left topology event itself and data nodes switch.
+         *
+         * @param dataNodesAutoAdjustScaleDown Timeout in seconds between node left topology event itself
+         *                                     and data nodes switch.
+         * @return This instance.
+         */
+        public Builder dataNodesAutoAdjustScaleDown(int dataNodesAutoAdjustScaleDown) {
+            this.dataNodesAutoAdjustScaleDown = dataNodesAutoAdjustScaleDown;
+
+            return this;
+        }
+
+        /**
+         * Builds the distribution zone configuration.
+         *
+         * @return Distribution zone configuration.
+         */
+        public DistributionZoneCfg build() {
+            if (dataNodesAutoAdjust != null
+                    && (dataNodesAutoAdjustScaleUp != null || dataNodesAutoAdjustScaleDown != null)
+                ) {
+                throw new IllegalArgumentException(
+                        String.format("Not compatible parameters [dataNodesAutoAdjust=%s, "
```

Comment:
I would add here some more info, like why this parameters are incompatible

Reply:
I think it's not needed. The builder is a part of internal api that will be used by SQL module. SQL module will check query before a builder creation. So actually this exception will never be thrown.